pkg/ruler/ruler.go

```diff
@@ -736,3 +740,26 @@ func (r *Ruler) Rules(ctx context.Context, in *RulesRequest) (*RulesResponse, er
 
 	return &RulesResponse{Groups: groupDescs}, nil
 }
+
+// AssertMaxRuleGroupsPerUser limit has not been reached compared to the current
+// number of total rule groups in input and returns an error if so.
+func (r *Ruler) AssertMaxRuleGroupsPerUser(userID string, rg int) error {
+	limit := r.limits.RulerMaxRuleGroupsPerUser(userID)
+
+	if rg < limit {
```

Comment:
Hitting the limit is still OK, so:

```suggestion
	if rg <= limit {
```

_Same comments applies to `AssertMaxRulesPerRuleGroupPerUser()` too_

Reply:
This is just as we're about to create a new one, e.g. In the case where limit is 3 and we have 2, the above would be true and would create the 3rd.